fault/system_verilog_target.py

```diff
@@ -0,0 +1,191 @@
+            if isinstance(action.port, m.ArrayType) and \
+                    not isinstance(action.port.T, m.BitKind):
+                return SystemVerilogTarget.generate_array_action_code(i, action)
+            name = verilator_utils.verilator_name(action.port.name)
+            if isinstance(action.value, BitVector) and \
+                    action.value.num_bits > 32:
+                raise NotImplementedError()
+            else:
+                value = action.value
+                if isinstance(action.port, m.SIntType) and value < 0:
+                    # Handle sign extension for verilator since it expects and
+                    # unsigned c type
+                    port_len = len(action.port)
+                    value = BitVector(value, port_len).as_uint()
+                return [f"{name} = {value};", "#5"]
+        if isinstance(action, actions.Print):
+            name = verilator_utils.verilator_name(action.port.name)
+            if isinstance(action.port, m.ArrayType) and \
+                    not isinstance(action.port.T, m.BitKind):
+                return SystemVerilogTarget.generate_array_action_code(i, action)
+            else:
+                return [f'$display("{action.port.debug_name} = '
+                        f'{action.format_str}", {name});']
+        if isinstance(action, actions.Expect):
+            # For verilator, if an expect is "AnyValue" we don't need to
+            # perform the expect.
+            if value_utils.is_any(action.value):
+                return []
+            if isinstance(action.port, m.ArrayType) and \
+                    not isinstance(action.port.T, m.BitKind):
+                return SystemVerilogTarget.generate_array_action_code(i, action)
+            name = verilator_utils.verilator_name(action.port.name)
+            value = action.value
+            if isinstance(value, actions.Peek):
+                value = f"{value.port.name}"
+            elif isinstance(action.port, m.SIntType) and value < 0:
+                # Handle sign extension for verilator since it expects and
+                # unsigned c type
+                port_len = len(action.port)
+                value = BitVector(value, port_len).as_uint()
+
+            return [f"if ({name} != {value}) $error(\"Failed on action={i} checking port {action.port.name}. Expected %x, got %x\", {value}, {name});"]
+        if isinstance(action, actions.Eval):
+            # Eval implicit in SV simulations
+            return []
+        if isinstance(action, actions.Step):
+            name = verilator_utils.verilator_name(action.clock.name)
+            code = []
+            for step in range(action.steps):
+                # code.append("top->eval();")
+                # code.append("main_time++;")
+                # code.append("#if VM_TRACE")
+                # code.append("tracer->dump(main_time);")
+                # code.append("#endif")
+                code.append(f"#5 {name} ^= 1;")
```

Comment:
why `#5`?

Reply:
Based on an example from @priyanka-raina https://github.com/priyanka-raina/fault_vcs_test/blob/master/verilog_tester_counter.v

I think this should also be a parameter. Basically, from what I understand, `#` indicates delay, and the number of timesteps to delay depends on your clock period. Using `#5` for each step indicates a clock period of 10 steps. Not sure what a good default is, but I think this should definitely be configurable by the user.